contracts/upgradeable_contracts/multi_amb_erc20_to_erc677/TokenProxy.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity 0.4.24;
+
+import "../../upgradeability/Proxy.sol";
+
+interface IPermittableTokenVersion {
+    function version() external pure returns (string);
+}
+
+/**
+* @title TokenProxy
+* @dev Helps to reduces the size of the deployed bytecode for automatically created tokens, by using a proxy contract.
+*/
+contract TokenProxy is Proxy {
+    // storage layout is copied from PermittableToken.sol
+    string internal name;
+    string internal symbol;
+    uint8 internal decimals;
+    mapping(address => uint256) internal balances;
+    uint256 internal totalSupply;
+    mapping(address => mapping(address => uint256)) internal allowed;
+    address internal owner;
+    bool internal mintingFinished;
+    address internal bridgeContractAddr;
+    // string public constant version = "1";
+    bytes32 internal DOMAIN_SEPARATOR;
+    // bytes32 public constant PERMIT_TYPEHASH = 0xea2aa0a1be11a07ed86d755c93467f4f82362b452371d1ba94d1715123511acb;
+    mapping(address => uint256) internal nonces;
+    mapping(address => mapping(address => uint256)) internal expirations;
+
+    /**
+    * @dev Creates a non-upgradeable token proxy for PermitableToken.sol, initializes its eternalStorage.
+    * @param _tokenImage address of the token image used for mirrowing all functions.
+    * @param _name token name.
+    * @param _symbol token symbol.
+    * @param _decimals token decimals.
+    * @param _chainId chain id for current network.
+    */
+    constructor(address _tokenImage, string memory _name, string memory _symbol, uint8 _decimals, uint256 _chainId)
+        public
+    {
+        // same as
+        // string memory version = IPermittableTokenVersion(_tokenImage).version();
+        // but compiles without errors for spuriosDragon
+        uint256 versionLen;
+        IPermittableTokenVersion(_tokenImage).version(); // makes a call to version()
+        assembly {
+            let tmp := mload(0x40) // get free temporary memory
+            returndatacopy(tmp, 32, 32) // save length of returned string to memory slot pointed by tmp
+            versionLen := mload(tmp) // save length of returned string to versionLen variable
+        }
+        string memory version = new string(versionLen);
+        assembly {
+            returndatacopy(add(version, 32), 64, versionLen) // copy string itself to version variable
```

Comment:
Once https://github.com/poanetwork/tokenbridge-contracts/pull/456 is merged, I think this code could be simplified to 
```
string memory version = IPermittableTokenVersion(_tokenImage).version();
```
as commented in the code

Reply:
Since #456 was merged, @k1rill-fedoseev could you address this comment?

Reply:
Yes, sure. A have updated this piece within the last commit.